src/coreclr/src/jit/importer.cpp

```diff
@@ -9083,7 +9127,19 @@ GenTree* Compiler::impFixupCallStructReturn(GenTreeCall* call, CORINFO_CLASS_HAN
     // and we change the return type on those calls here.
     //
     structPassingKind howToReturnStruct;
-    var_types         returnType = getReturnTypeForStruct(retClsHnd, &howToReturnStruct);
+    var_types         returnType;
+
+#if defined(TARGET_WINDOWS) && !defined(TARGET_ARM)
+    if (call->gtCallMoreFlags & GTF_CALL_M_UNMGD_INST_CALL)
```

Comment:
I think it should be 
```
if (call->gtCallMoreFlags & GTF_CALL_M_UNMGD_INST_CALL == 0)
{
    // managed call;
    howToReturnStruct = SPK_ByReference;
    returnType        = TYP_UNKNOWN;
}
```

Reply:
No this part is correct. This is the enforcement of the instance-call modifications to the Windows calling convention.